### PR TITLE
Added accessibility section

### PR DESCRIPTION
**Pull request by vmpstr**
Please take a look. Rendered output: https://vmpstr.github.io/display-locking/#accessibility

/cc @alice

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -165,6 +165,42 @@ the user-agent should retain the previously computed layout state if possible.
...
 
+Accessibility {#accessibility}
+=============
+
+Similar to the way 'subtree-visibility' affects painted output of the element's
+subtree, it also affects information exposed to the
+<a href="https://w3c.github.io/css-aam/#dfn-accessibility-tree">accessibility
+tree</a> and <a href="https://w3c.github.io/css-aam/#dfn-assistive-technology">
+assistive technologies</a>:
+* The structure of the page should be exposed to assistive technologies.
```
Comment (alice) [inline index.bs]: Perhaps flesh this out a little more, something like

* Subtrees with a `subtree-visibility` value of `auto` or `visible` should be included in the accessibility tree as usual.
* Subtrees with a `subtree-visibility` value of `skipped` should be excluded from the accessibility tree.

File: index.bs [hunk 2]
```diff
@@ -165,6 +165,42 @@ the user-agent should retain the previously computed layout state if possible.
...
+* When [=skipped=], the element's subtree should be exposed as
+    <a href="https://w3c.github.io/css-aam/#dfn-hidden">hidden</a>.
+
+<div class=note>
+  The general idea for exposing elements as hidden is the following: if content
```
Comment (alice) [inline index.bs]: Suggested rephrasing:

Since assistive technologies may provide quick access to offscreen elements, it is important that elements in `subtree-visibility: auto` subtrees be made available to assistive technologies to provide this functionality, since they are intended to be observable to users.

Conversely, since elements in `subtree-visibility: hidden` subtrees are not intended to be observable by users, they should not be exposed to assistive technology.

File: index.bs [hunk 3]
```diff
@@ -165,6 +165,42 @@ the user-agent should retain the previously computed layout state if possible.
...
+
+<div class=note>
+  The requirement of rendering performance equivalency stems from privacy
+  considerations. It is imperative that the user-agent ensures that a page
+  using 'subtree-visibility' cannot use timing information to deduce whether
+  the user is using assistive technologies. For this reason, the rendering
+  performance of information exposed to assistive technologies must be the same
+  as the rendering performance of information exposed to painted output.
+
+  In other words, if the user-agent omits rendering work when producing painted
```
Comment (alice) [inline index.bs]: This is nicely worded.

File: index.bs (hunk 2 above)
Comment (vmpstr) [reply index.bs]: This particular phrasing seems to contradict the normative text. The problem is specifically about the "auto" value. When off-screen, the elements are not rendered (ie they are 'skipped'). So they should be excluded from the accessibility tree (according to the bullet points above).

Here, however, we are saying that should be included for quick off-screen access. I think what's missing is the distinction between exposing the element vs exposing the layout information (size and positioning). 

I think we should expose the element (which is what I meant by the 'structure of the page'). But we can't expose the layout information, since that would cause rendering to me noticeably slower. 

Do you know if there's terminology here that we can use to convey this?

Comment (alice) [reply index.bs]: > When off-screen, the elements are not rendered (ie they are 'skipped'). So they should be excluded from the accessibility tree (according to the bullet points above).

Which bullet points?

> I think we should expose the element (which is what I meant by the 'structure of the page'). 

Hm, I found that wording hard to follow. It's ambiguous, I think.

> But we can't expose the layout information, since that would cause rendering to me noticeably slower.

Agreed.

> Do you know if there's terminology here that we can use to convey this?

I thought it was covered by the paragraph below.

Comment (vmpstr) [reply index.bs]: > > When off-screen, the elements are not rendered (ie they are 'skipped'). So they should be excluded from the accessibility tree (according to the bullet points above).
> 
> Which bullet points?

Sorry, I was referencing the bullet points you suggested in the previous comment. Specifically the second one says:

* Subtrees with a subtree-visibility value of skipped should be excluded from the accessibility tree.

"skipped" as we defined it in the spec means not-rendered: this would be, among others, the off-screen elements with `subtree-visibility: auto`. So, that bullet point says don't include off-screen elements with `subtree-visibility: auto` in the accessibility tree. 

However, the suggested phrasing of this paragraph says that off-screen elements with `subtree-visibility: auto` should be made available to the assistive technologies.

Doesn't that mean we should include them in the accessibility tree? Or is there a distinction between including something in the accessibility tree vs making it available to assistive technology?

File: index.bs (hunk 3 above)
Comment (vmpstr) [reply index.bs]: Thanks! :)

File: index.bs (hunk 1 above)
Comment (vmpstr) [reply index.bs]: Done.

File: index.bs (hunk 2 above)
Comment (vmpstr) [reply index.bs]: I've rephrased this a bit, new text: https://vmpstr.github.io/display-locking/#accessibility 

Let me know if that reads ok, or whether I should make more changes. I'd rather get this text correct and have it be unambiguous

File: index.bs [hunk 4]
```diff
@@ -165,6 +165,60 @@ the user-agent should retain the previously computed layout state if possible.
...
+Accessibility {#accessibility}
+=============
+
+Similar to the way 'subtree-visibility' affects painted output of the element's
+subtree, it also affects information exposed to the
+<a href="https://w3c.github.io/css-aam/#dfn-accessibility-tree">accessibility
+tree</a> and <a href="https://w3c.github.io/css-aam/#dfn-assistive-technology">
+assistive technologies</a>:
+* Subtrees affected by ''subtree-visibility: visible'' or ''subtree-visibility: auto''
+    and they are not [=skipped=] should be included in the accessibility tree as usual.
```
Comment (chrishtr) [inline index.bs]:
```suggestion
    that are not [=skipped=] should be included in the accessibility tree as usual.
```

File: index.bs [hunk 5]
```diff
@@ -165,6 +165,60 @@ the user-agent should retain the previously computed layout state if possible.
...
+* Subtrees of [=skipped=] elements not be included in the accessbility tree,
```
Comment (chrishtr) [inline index.bs]:
```suggestion
* Subtrees of [=skipped=] elements should not be included in the accessibility tree,
```

File: index.bs [hunk 6]
```diff
@@ -165,6 +165,60 @@ the user-agent should retain the previously computed layout state if possible.
...
+  The requirement of rendering performance equivalency stems from privacy
+  considerations. It is imperative that the user-agent ensures that a page
+  using 'subtree-visibility' cannot use timing information to deduce whether
+  the user is using assistive technologies. For this reason, the rendering
+  performance of information exposed to assistive technologies must be the same
+  as the rendering performance of information exposed to painted output.
+
+  In other words, if the user-agent omits rendering work when producing painted
+  output, then it must also omit the same rendering work when providing
+  information for assistive technologies.
```
Comment (chrishtr) [inline index.bs]:
```suggestion
  information for assistive technologies, and vice-versa.
```

**Comment by vmpstr** [comment]
@alice could you please take another look when you get a chance?

File: index.bs (hunk 2 above)
Comment (chrishtr) [reply index.bs]: I think the new phrasing looks great! Let's commit this PR. Alice, if you have more comments please add them and we can add them in a new PR.

**Comment by vmpstr** [comment]
Committed per chrishtr's comment; we can adjust in another PR if needed.
